### PR TITLE
Set the securityContext for the HTTPD Apache instrumentation

### LINKED ISSUE #2050: SecurityContext is not propagated to apache-httpd instrumentation init containers.
If a security context is set for the init containers when using the apache-httpd instrumentation, the securityContext is not propagated to its `initContainers`.

 As a result, the instrumented pod cannot be started because the `otel-agent-attach-apache` cannot copy the content from the `/otp/opentelemetry`. folder

### PR DESCRIPTION
**Pull request by iblancasa**
Resolves #2050 .

### CODE REVIEW

File: pkg/instrumentation/apachehttpd.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ func injectApacheHttpdagent(_ logr.Logger, apacheSpec v1alpha1.ApacheHttpd, pod
 			Command: []string{"/bin/sh", "-c"},
 			Args: []string{
 				// Copy agent binaries to shared volume
-				"cp -ar /opt/opentelemetry/* " + apacheAgentDirFull + " && " +
```
Comment (pavolloffay) [inline pkg/instrumentation/apachehttpd.go:148]: @chrlic could you please review?

**Comment by iblancasa** [comment]
#2069 is a blocker for this PR.

**Comment by iblancasa** [comment]
Blocked by #2225

**Comment by pavolloffay** [comment]
CI failed

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**